engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -2003,18 +2006,19 @@ - (void)performOrientationUpdate:(UIInterfaceOrientationMask)new_preferences {
     self.orientationPreferences = new_preferences;
 
     if (@available(iOS 16.0, *)) {
-      NSSet<UIScene*>* scenes =
-#if APPLICATION_EXTENSION_API_ONLY
-          self.flutterWindowSceneIfViewLoaded
-              ? [NSSet setWithObject:self.flutterWindowSceneIfViewLoaded]
-              : [NSSet set];
-#else
-          [UIApplication.sharedApplication.connectedScenes
-              filteredSetUsingPredicate:[NSPredicate predicateWithBlock:^BOOL(
-                                                         id scene, NSDictionary* bindings) {
-                return [scene isKindOfClass:[UIWindowScene class]];
-              }]];
-#endif
+      UIApplication* flutterApplication = [FlutterSharedApplication uiApplication];
+      NSSet<UIScene*>* scenes;
```

Comment:
Suggestion to guarantee we never hand a nil `scenes` to `requestGeometryUpdateForWindowScenes:`:
```suggestion
      NSSet<UIScene*>* scenes = [NSSet set];
```